search-app/src/cmr/search/services/query_execution/facets/hierarchical_v2_facets.clj

```diff
@@ -79,6 +81,14 @@
     {subfield {:terms {:field (str (name field) "." (name subfield))
                        :size size}
                :aggs (merge {:coll-count frf/collection-count-aggregation}
+                            (cond
+                              (and (= field :science-keywords-humanized)
+                                   (or (= subfield :term)
+                                       (= subfield :variable-level-1)
+                                       (= subfield :variable-level-2))) (hierarchical-aggregation-builder field [:detailed-variable] size)
```

Comment:
With this logic, will the aggregation ever include `variable-level-1` or `variable-level-2` if the science keyword hierarchy is `term -> variable-level-1 -> variable-level-2`?

Reply:
Yes, to view what the aggregation looks like see expected-nested-facet-aggregations-for-science-keywords expected test result in the cmr.search.test.services.query-execution.facets.hierarchical-v2-facets name space.  We query for all buckets in 1 query.

```
 Category
    Topic
        Term
            Detailed-Variable
            Variable-Level-1
                Detailed-Variable
                Variable-Level-2
                    Detailed-Variable
                    Variable-Level-3
                        Detailed-Variable

```
Elastic search will fill in what it has.
Then when we parse the result we first walk down the hierarchy Category, Topic, Term, Variable-Level-1, Variable-Level-2, Variable-Level-3, Detailed-Variable. At any point down the traversal a bucket could be empty.  For example if Variable-Level-3 is empty, then we traverse Category, Topic, Term, Variable-Level-1, Variable-Level-2, Variable-Level-3. since the Variable-Level-3 bucket is empty we check the Detailed-Variable bucked under Variable-Level-2 to see it it exists.  If it does then the record has Category, Topic, Term, Variable-Level-1, Variable-Level-2, Detailed-Variable.  So the facets get built in that manor. 
    

Below is a sample of the aggregations starting from Term through Variable-Level-1 where you can see that we also include Detailed-Variable.

{:terms {:field "science-keywords-humanized.term", :size 1},
         :aggs
         {:coll-count {:reverse_nested {},
                       :aggs {:concept-id {:terms {:field :concept-id, :size 1}}}},
          :detailed-variable
          {:terms {:field "science-keywords-humanized.detailed-variable", :size 1},
           :aggs
           {:coll-count {:reverse_nested {},
                         :aggs {:concept-id {:terms {:field :concept-id, :size 1}}}}}},
          :variable-level-1
          {:terms {:field "science-keywords-humanized.variable-level-1", :size 1},
           :aggs
           {:coll-count {:reverse_nested {},
                         :aggs {:concept-id {:terms {:field :concept-id, :size 1}}}},